chore: add AI agent involvement to PR template (#21812) (#21813)
pingcap/docs-cn: .agents/skills/docs-pr-metadata-guard/SKILL.md

Before changing a PR body, read `.github/pull_request_template.md`.

## Workflow

1. PR titles follow `pingcap/community` commit-message style (for example, `Fix typos in tidb-monitoring-api.md`). Titles and section content can be in Chinese or English; keep product names, commands, and paths in English.
2. For a new PR, start from `.github/pull_request_template.md` instead of writing the body from scratch.
   - Copy the template into a local Markdown file and fill in the mutable fields.
   - Submit with `gh pr create --body-file <local-file>`. This is the reliable method. The `-T` / `--template` flag matches a template by **name** (the basename, for example `gh pr create -T pull_request_template.md`), not by path, so `-T .github/pull_request_template.md` will not match. Prefer `--body-file`.
   - Review the local file against the template before calling `gh`.
3. Fill in the required sections with concrete information.
   - **What is changed, added or deleted? (Required)**: describe what changed and why in clear, specific language. Do not leave this blank or fill it with a generic placeholder.

>    - **Which TiDB version(s) do your changes apply to? (Required)**: check at least one version checkbox. Follow the affected-version rules in `.agents/shared/repo-conventions.md` and `CONTRIBUTING.md` (版本选择指南):

Reviewer: ```suggestion
   - **Which TiDB version(s) do your changes apply to? (Required)**: check at least one version checkbox. Follow the `CONTRIBUTING.md` (版本选择指南):
```

   - **AI agent involvement**: when this section is present in the template, keep it intact. Check its checkbox only when the changes were primarily made by an AI agent on behalf of the PR author; otherwise leave it unchecked.
   - **Do your changes match any of the following descriptions?**: check all that apply. If the change needs different wording on another branch, check `Need modification after applied to another branch` and comment `/label requires-version-specific-changes` to trigger the bot. (Note: the PR template still prints the old command `/label version-specific-changes-required`, which the bot now rejects because the actual label is `requires-version-specific-changes`. Use the actual label name.)
4. Choose the correct base branch.
   - Default to `master` for most documentation PRs.
   - Use a specific `release-X.Y` branch when the change is scoped to a single published version and does not apply to `master`.

>    - Do not open PRs for the Chinese .agents/` content: it is auto-translated weekly from `pingcap/docs`. Fix the English source instead.

Reviewer: ```suggestion
   - Do not open PRs for the  `/ai` docs and `/tidb-cloud` docs: they are auto-translated weekly from `pingcap/docs`. Fix the English source instead.
```

>    - Do not open PRs for the Chinese .agents/` content: it is auto-translated weekly from `pingcap/docs` (see `.agents/shared/repo-conventions.md`). Fix the English source instead.

Reviewer: ```suggestion
   - Do not open PRs for the Chinese .agents/` content: it is auto-translated weekly from `pingcap/docs`. Fix the English source instead.
```
